docs/content/overriding-styles.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: Overriding styles with the sx prop
+---
+
+Our goal with Primer Components is to hit the sweet spot between providing too little and too much styling flexibility; too little and the design system is too rigid, and too much and it becomes too difficult to maintain a consistent style. Our components already support a standard set of [system props](/system-props), but sometimes a component just isn't *quite* flexible enough to look the way you need it to look. For those cases, we provide the `sx` prop.
+
+The `sx` prop allows ad-hoc styling that is still theme aware. Declare the styles you want to apply in camel-cased object notation, and any values use appropriate theme values; for example, any `color` property values will use the values from the `colors` key in the theme.
+
+## Basic example
+
+This example demonstrates applying a bottom border to `Heading`, a component that does not receive `BORDER` system props. The `borderBottomWidth` value comes from `theme.borderWidths` and `borderBottomColor` comes from `theme.colors`.
+
+```jsx live
+<Heading pb={2}>Heading</Heading>
+
+<Heading
+  pb={2}
+  sx={{
+    borderBottomWidth: 1,
+    borderBottomColor: 'border.gray',
```

Comment:
Does the `sx` prop look up values in the theme in the same exact why that styled system does? Have we double checked that pulling from the theme works with 1) custom themes 2) when a user doesn't have a ThemeProvider at all (does it fall back to the default theme?)

We might want to add some documentation about the above two points. Something like "If you have a custom theme set up with a ThemeProvider, the custom theme will be honored". We might also want to point people to the styled-system docs for a refresher on how pulling from the theme works? (e.g. color related properties pull from the color object, spacing related properties pull from the space object, etc) This can be kind of confusing to some people. Maybe those docs should live in the theming docs or core concepts, and we just link to them from here.

Reply:
Yes, it works just like styled-system, and it seems to be used in some of styled-system's internals. Here's a quick test I did trying out a few of the theme forms:

![image](https://user-images.githubusercontent.com/189606/80411987-fa511d00-8881-11ea-8f02-1e9193b9b20f.png)

I'll add a note to the docs.